Crash: cast float to non-existent enum

```c
void foo(void) {
    (enum E)2.0;
}
```

```
panic: attempt to use null value
/Users/ehaas/source/arocc/src/aro/Value.zig:174:57: 0x1084a7b40 in floatToInt (arocc)
    const bits: usize = @intCast(dest_ty.bitSizeof(comp).?);
                                                        ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:5962:43: 0x108614be4 in castType (arocc)
                _ = try res.val.floatToInt(to, p.comp);
                                          ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:6704:29: 0x108612460 in castExpr (arocc)
        try operand.castType(p, ty, operand_tok, l_paren);
                            ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:6597:29: 0x108608285 in mulExpr (arocc)
    var lhs = try p.castExpr();
                            ^
```

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -5962,6 +5962,10 @@ pub const Result = struct {
             if (to.is(.bool)) {
                 res.val.boolCast(p.comp);
             } else if (old_float and new_int) {
+                if (to.hasIncompleteSize()) {
+                    try p.errStr(.cast_to_incomplete_type, l_paren, try p.typeStr(to));
+                    return error.ParsingFailed;
+                }
                 // Explicit cast, no conversion warning
                 _ = try res.val.floatToInt(to, p.comp);
             } else if (new_float and old_int) {

--- a/test/cases/casts.c
+++ b/test/cases/casts.c
@@ -16,6 +16,7 @@ void foo(void) {
     (enum E)1;
     int *ptr;
     ptr = (int *)(void)5;
+    (enum DoesNotExist)2.0;
 }
 
 #define EXPECTED_ERRORS "casts.c:5:5: error: cannot cast to non arithmetic or pointer type 'struct Foo'" \
@@ -27,4 +28,5 @@ void foo(void) {
     "casts.c:11:13: error: pointer cannot be cast to type 'float'" \
     "casts.c:16:5: error: cast to incomplete type 'enum E'" \
     "casts.c:18:18: error: used type 'void' where arithmetic or pointer type is required" \
+    "casts.c:19:5: error: cast to incomplete type 'enum DoesNotExist'" \
 

